test/testdrive/kafka-compression.td

```diff
@@ -47,23 +47,3 @@ world
 > SELECT text FROM lz4
 hello
 world
-
```

Comment:
just curious, why is this stuff getting moved to its own file?

Reply:
zstd compression is the one feature we were testing for in the Kafka version test that isn't supported back to 4.x/5.x. Right now we feed all testdrive files prefixed with 'kafka-*' into the version test - so moving this into its own zstd- file is a slightly hacky way of excluding it from the version suite while making sure that new kafka tests are picked up by the version suite.

Ideally we would have a more expressive way of feeding test files into this or marking version compat, but...